idaes/apps/caprese/examples/cstr_rodrigo/cstr_rodrigo_model2.py

```diff
@@ -0,0 +1,174 @@
+    m.V = Param(initialize=100)
+    m.UA = Param(initialize=20000 * 60)
+    m.rho = Param(initialize=1000)
+    m.Cp = Param(initialize=4.2)
+    m.Vw = Param(initialize=10)
+    m.rhow = Param(initialize=1000)
+    m.Cpw = Param(initialize=4.2)
+    m.k0 = Param(initialize=4.11e13)
+    m.E = Param(initialize=76534.704)
+    m.R = Param(initialize=8.314472)
+    m.Er = Param(initialize=lambda m: (value(m.E) / value(m.R)))
+    m.dH = Param(initialize=596619.)
+
+    m.F = Param(m.t, mutable=True, default=1.2000000000000000E+02)
+    m.Fw = Param(m.t, mutable=True, default=3.0000000000000000E+01)
+    
+    #States
+    m.Ca = Var(m.t,  initialize=1.60659680385930765667001907104350E-02)
+    m.Tall = Var(m.t, m.T_ind)
+    
+    #Algebraic var
+    m.k = Var(m.t,  initialize=4.70706140E+02)
+    
+    #Controls
+    m.Tjinb = Var(m.t, initialize=250)
+    
+    #Derivative variables
+    if steady:
+        m.Cadot = Var(m.t, initialize = 0.0)
+        m.Talldot = Var(m.t, m.T_ind, initialize = 0.0)
+        m.Cadot.fix()
+        m.Talldot.fix()
+    else:
+        m.Cadot = DerivativeVar(m.Ca, initialize=-3.58709135E+01)
+        m.Talldot = DerivativeVar(m.Tall, wrt = m.t)
+        
+    #Constraint rules
+    def _rule_k(m, i):
+        return m.k[i] == m.k0 * exp(-m.Er / m.Tall[i, "T"])
+
+    def _rule_ca(m, i):
+        rule = (m.Cadot[i] == (m.F[i] / m.V) * 
+                              (m.Cainb - m.Ca[i]) - 2 * 
+                              m.k[i] * 
+                              m.Ca[i] ** 2)
+        return rule
+
+    def _rule_t(m, i):
+        return m.Talldot[i, "T"] == ((m.F[i] / m.V) * (m.Tinb - m.Tall[i, "T"]) + 
+                                     2.0 * m.dH / (m.rho * m.Cp) * m.k[i] * m.Ca[i] ** 2 -
+                                     m.UA / (m.V * m.rho * m.Cp) * (m.Tall[i, "T"] - m.Tall[i, "Tj"]))
+
+    def _rule_tj(m, i):
+        return m.Talldot[i, "Tj"] == ((m.Fw[i] / m.Vw) * (m.Tjinb[i] - m.Tall[i, "Tj"]) + 
+                                      m.UA/(m.Vw * m.rhow * m.Cpw) * (m.Tall[i, "T"] - m.Tall[i, "Tj"]))
+    
+    #Constraints
+    m.kdef = Constraint(m.t, rule = _rule_k)
+    m.de_ca = Constraint(m.t, rule = _rule_ca)
+    m.de_T = Constraint(m.t, rule = _rule_t)
+    m.de_Tj = Constraint(m.t, rule = _rule_tj)
+        
+    # Time discretization
+    if not steady:
+        disc = TransformationFactory('dae.collocation')
+        disc.apply_to(m, wrt=m.t, nfe=ntfe, ncp=ntcp, scheme='LAGRANGE-RADAU')
+        
+    # Set initialize values for IndexedVar with string index
+    m.Tall[:, "T"] = 3.92336059452774350120307644829154E+02
+    m.Tall[:, "Tj"] = 3.77995395658401662331016268581152E+02
+    m.Talldot[:, "T"] = 5.19191848E+03
+    m.Talldot[:, "Tj"] = -9.70467399E+02
+    
+    #Set initial conditions for states
+    if not steady:
+        m.Ca[0].fix(1.9193793974995963E-02)
+        m.Tall[0, "T"].fix(3.8400724261199036E+02)
+        m.Tall[0, "Tj"].fix(3.7127352272578315E+02)
+    

```

Comment:
Same comments as above for initial guesses.

Reply:
Thanks for catching these! I will update them.